code/client/installhelper

```diff
@@ -0,0 +1,372 @@
+    '''Returns a dict containing any logged in users uid and username'''
+
+    # Var declaration
+    user_details = {}
+
+    # Retrieve details of all running processes
+    cmd = ['/bin/ps', '-axo', 'uid,login,args']
+    cmd_out = subprocess.run(cmd, capture_output = True, check = False)
+    # Split out the above output
+    for some_process in cmd_out.stdout.decode().splitlines():
+        # Retrieve details for users that are running loginwindow.app, excluding root
+        if "loginwindow.app" in some_process.lower() and some_process.split()[0] != "0":
+            # Capture the userid and username for all users running the loginwindow
+            user_details[some_process.split()[0]] = some_process.split()[1]
+
+    # Returns either an empty dict, or a dict with userid and usernames
+    return user_details
+
+
+def log(msg):
+    '''Logs messages from this tool with an identifier'''
+
+    munkilog.log("%s-%s: %s" % (INSTALL_HELPER, PASSED_ARG, msg))
+
+
+def is_managedsoftwareupdate_running():
+    '''If managedsoftwareupdate is running, check again in x seconds'''
+
+    # Check to see if managedsoftwareupdate is running
+    managedsoftwareupdate_pid = osutils.pythonScriptRunning(PROG_NAME)
+
+    # An instance of managedsoftwareupdate is running, so we need to try again later
+    while managedsoftwareupdate_pid:
+        log("*" * 60)
+        log("%s is running as pid %s." % (managedsoftwareupdate_pid,  os.getpid()))
+        log("checking again in %s seconds" % CHECK_TIME)
+        log("*" * 60)
+        time.sleep(CHECK_TIME)
+
+    # If managedsoftwareupdate is not running.. return from function to proceed
+    log("%s is not running, proceeding..." % (PROG_NAME))
+
+
+def reload_launch_daemons(launch_daemons):
+    '''Reloads Munki's Launch Daemons'''
+
+    # Log what we're upto
+    log("Reloading Launch Daemons")
+
+    # Retrieve all running launch daemons
+    cmd = ['/bin/launchctl', 'list']
+    cmd_out = subprocess.run(cmd, capture_output = True, check = False)
+    launchd_labels = split_launchctl_list(cmd_out)
+
+    # For each plist in /Library/LaunchDaemons/ prefixed with LAUNCHD_PREFIX
+    for launch_daemon in launch_daemons:
+        try:
+            # Make sure content is valid, before proceeding to reload
+            daemon_content = FoundationPlist.readPlist(launch_daemon)
+        except FoundationPlist.NSPropertyListSerializationException:
+            break
+        # Get the Label
+        daemon_label = daemon_content.get('Label')
+        # If we have a label
+        if daemon_label:
+            # Check to see if the launch daemon is loaded, before stopping
+            if daemon_label in launchd_labels:
+                # Stop the launch daemon at the login window
+                log("Stopping: %s" % launch_daemon)
+                cmd = ['/bin/launchctl', 'bootout', 'system/' + daemon_label]
+                subprocess.call(cmd)
+            # Load the launch daemon at the login window
```

Comment:
Feels like this was just copied from elsewhere as there's nothing specific to the loginwindow here.

Reply:
Removed erroneous login window comments in - https://github.com/macmule/munki/commit/3a9302164f2ead8103e32ae94278de743c8dae22